src/com/facebook/buck/zip/JarBuilder.java

```diff
@@ -235,6 +254,20 @@ private void addEntryToJar(JarEntrySupplier entrySupplier, CustomJarOutputStream
       return;
     }
 
+    // Collect all services together for later merging and addition to the output jar
+    if (isService(entryName)) {
+      try (InputStream entryInputStream = entrySupplier.getInputStreamSupplier().get()) {
+        if (entryInputStream != null) {
+          // Null stream means a directory
```

Comment:
Minor: If you have `isService` also confirm that the name does not end in `/` (i.e., is not a directory) you can `Preconditions.checkNotNull` in the initializer rather than having this check.